config/v1/types_infrastructure.go

```diff
@@ -630,7 +630,7 @@ const (
 )
 
 // GCPServiceEndpointName is the name of the GCP Service Endpoint.
-// +kubebuilder:validation:Enum=Compute;Container;CloudResourceManager;DNS;File;IAM;ServiceUsage;Storage;TagManager
+// +kubebuilder:validation:Enum=Compute;Container;CloudResourceManager;DNS;File;IAM;ServiceUsage;Storage
```

Comment:
What happens to existing resources when you remove this value? This API has shipped so to be able to remove a value we would need to be certain that it won't break existing resources.

Can you please add a ratcheting validation test (see `tests/README.md`) to make sure this change will allow existing resources with this value set to:
* Update the field itself to remove the bad value
* Update the field value to add a new value
* Update other fields around the field while leaving an invalid value in place